scxml_core/src/state_machine.cpp

```diff
@@ -630,11 +636,15 @@ bool StateMachine::addPreconditionCallback(const std::string& st_name, Precondit
     return false;
   }
 
-  if (precond_callbacks_.count(st_name) > 0)
+  auto pc = precond_callbacks_.find(st_name);
+  if (pc == precond_callbacks_.end())
```

Comment:
I feel that you still could've use the previous if statement since the type is still a map with state name strings as the keys

Reply:
The iterator `pc` is needed at 646 in order to add additional callbacks if the state is already in the `precond_callbacks_` map. Reusing the iterator to determine if the state is in the map struck me as more efficient than having a separate check using `count`.

It would be possible to replace 639 through 647 with something more brief using the result returned by `insert`, but I don't think it's as reader-friendly:
```
auto result = precond_callbacks_.insert(std::make_pair(st_name, std::vector<PreconditionCallback>({ cb })));
if (!result.second)
  result.first->second.push_back(cb);
```